src/makeInfinityAdder.js

```diff
@@ -6,7 +6,25 @@
  */
 
 function makeInfinityAdder() {
-  // write code here
+  let result = 0;
+
+  function adder(number) {
+    if (typeof number !== 'undefined') {
+      result += number;
+    }
+
+    if (typeof number === 'undefined') {
+      const temp = result;
+
+      result = 0;
+
+      return temp;
+    }
+
+    return adder;
+  }
+
+  return adder;
```

Comment:
You can simplify your code a bit, deleting the second **if** and putting `return adder` into the first **if**

```suggestion
let result = 0;

  function adder(number) {
    if (typeof number !== 'undefined') {
      result += number;

      return adder;
    }

    const temp = result;

    result = 0;

    return temp;
  }

  return adder;
```

Reply:
agree